### DISCUSSION TITLE
room.setProperties() desync problem

### DISCUSSION THREAD

**Discussion by minstudio02** [wxyz-abcd/node-haxball#45] [Q&A]
room.setProperties() function cause a desync problem for clients who are already in the room. (when changing room name)
How can I fix that problem??

**Comment by wxyz-abcd** [comment]
It cannot be fixed completely. basro did not make an event for it in his default Haxball client. However, you can create a CustomEvent for RoomPropertiesChanged event and send/receive it in your own rooms. Note that this method only works for clients that are using your modified Haxball client, so it's important to create your own website and invite people to use it instead of original Haxball website if you want this to work correctly. The new room name will be applied to the room list in probably under 20 seconds. You may also increase the version number of the room, so that people can not join the room from original Haxball website. The people that come to your room from the original Haxball website will always be desynced unless basro changes its logic. There's nothing I or you can do about it.
